docs/sql-migration-guide.md

```diff
@@ -22,6 +22,10 @@ license: |
 * Table of contents
 {:toc}
 
+## Upgrading from Spark SQL 3.5 to 4.0
+
+- Since Spark 4.0, the default value of `spark.sql.maxSinglePartitionBytes` is changed from `Long.MaxValue` to `128m`.
```

Comment:
Thank you for moving to 4.0. Although it's obvious, we still need a directional guide to restore. Please see my previous comment, https://github.com/apache/spark/pull/42639#discussion_r1303993069.

Reply:
What I'm asking is adding the following, @pan3793 .
```
To restore the previous behavior, set `spark.sql.maxSinglePartitionBytes` to `Long.MaxValue`.
```

Reply:
@dongjoon-hyun how about now? Since the string literal `Long.MaxValue` does not work.

```
... to `9223372036854775807`(the value of `Long.MaxValue`).
```